### PR TITLE
Reformat timeseries metadata

### PR DESCRIPTION
**Pull request by daniel-thom**
This PR implements recent discussions about refactoring forecasts.  I made changes to the test data here:  https://github.com/NREL/PowerSystemsTestData/commit/7873078f0b149bdbd30644fd2ab1d40f70780542.  That is in a branch until review is complete.

If I get forecast-split functionality implemented before you review this then I'll add it.  But I will likely not finish that until next week.

### CODE REVIEW

File: src/PowerSystems.jl [hunk 1]
```diff
@@ -31,6 +31,7 @@ export PhaseShiftingTransformer
 export Forecast
 export Deterministic
 export Probabilistic
+export TimeseriesMetadata
```
Comment (jd-lara) [inline src/PowerSystems.jl]: Nit: MetaData.

File: src/parsers/forecast_parser.jl [hunk 2]
```diff
@@ -1,16 +1,60 @@
+
+const CATEGORY_STR_TO_COMPONENT = Dict{String, DataType}(
+    "Bus" => Bus,
+    "Generator" => Generator,
+    "Reserve" => Service,
+    "LoadZone" => LoadZones,
+    "ElectricLoad" => ElectricLoad,
```
Comment (jd-lara) [inline src/parsers/forecast_parser.jl]: Why is the referencing made to Abstract Types? I thought we had a default concrete struct to refer to.

File: src/parsers/forecast_parser.jl [hunk 3]
```diff
@@ -1,16 +1,60 @@
...
+)
+
+"""Describes how to construct forecasts from raw timeseries data files."""
+mutable struct TimeseriesMetadata
```
Comment (jd-lara) [inline src/parsers/forecast_parser.jl]: This is cool! I would name it something that is more descriptive of the purpose like `TimeSeriesFilesMetaData` or something that uses the word files.

File: src/parsers/forecast_parser.jl [hunk 4]
```diff
@@ -22,44 +66,115 @@ function ForecastInfos()
...
 
 """
-    forecast_csv_parser!(sys::System,
-                         directory_or_file::AbstractString,
-                         simulation="Simulation",
-                         category::Type{<:Component}=Component,
-                         label="scalingfactor";
-                         resolution=nothing,
-                         kwargs...)
+    add_forecasts!(sys::System, metadata_file::AbstractString; resolution=nothing)
```
Comment (jd-lara) [inline src/parsers/forecast_parser.jl]: We should clarify a bit more the expected use case here is when the user wants to parse directly from files

File: src/parsers/timeseries_formats.jl [hunk 5]
```diff
@@ -112,7 +112,7 @@ end
 
 This version of the function only has component_name to match the interface. It is unused.
 """
-function read_time_array(
+function read_timeseries(
```
Comment (jd-lara) [inline src/parsers/timeseries_formats.jl]: Also here, I would suggest renaming to `read_timeseries_file` or something more descriptive.

File: test/readforecastdata.jl [hunk 6]
```diff
@@ -1,5 +1,8 @@
+import DataFrames
 import TimeSeries
 
+const PS = PowerSystems
```
Comment (jd-lara) [inline test/readforecastdata.jl]: Nit: use PSY.

File: src/PowerSystems.jl (hunk 1 above)
Comment (daniel-thom) [reply src/PowerSystems.jl]: Disagree.  https://www.merriam-webster.com/dictionary/metadata?utm_campaign=sd&utm_medium=serp&utm_source=jsonld

File: src/parsers/timeseries_formats.jl (hunk 5 above)
Comment (daniel-thom) [reply src/parsers/timeseries_formats.jl]: I'm following conventions from many other places like pandas:
pandas.read_csv()
pandas.read_json()

File: src/parsers/forecast_parser.jl (hunk 3 above)
Comment (daniel-thom) [reply src/parsers/forecast_parser.jl]: Fixed.

File: src/parsers/forecast_parser.jl (hunk 4 above)
Comment (daniel-thom) [reply src/parsers/forecast_parser.jl]: Let's have a team discussion on whether we create many methods on the same function or create more unique function names.

File: test/readforecastdata.jl (hunk 6 above)
Comment (daniel-thom) [reply test/readforecastdata.jl]: Fixed.

File: deps/build.jl [hunk 7]
```diff
@@ -2,4 +2,4 @@
 include(joinpath(@__DIR__, "../src/utils/data.jl"))
 import .UtilsData: TestData
 
-download(TestData; branch = "improve-timeseries")
+download(TestData; branch = "forecast-metadata")
```
Comment (daniel-thom) [inline deps/build.jl]: Note this change.  Before merging to dev we need to merge the PowerSystemTestData branch into improve-timeseries.

Comment (jd-lara) [reply deps/build.jl]: Noted.

**Pull request merged**